webui/react/src/hooks/useSettings.ts

```diff
@@ -104,10 +105,7 @@ const queryToSettings = <T>(config: SettingsConfig<T>, query: string) => {
            */
           if (Array.isArray(paramValue)) {
             queryValue = paramValue.reduce<Primitive[]>((acc, value) => {
-              const parsedValue = queryParamToType<T>(baseType, value);
-
-              if (parsedValue !== undefined) acc.push(parsedValue);
-
```

Comment:
does this not work if we parse the value?

Reply:
the initial way this code was written is confused, it iterates over the values in `paramValue.reduce`, then tries to convert each string to the original `baseType` (which is `undefined || RunState[]`, not an individual state value)

`queryParamToType` always expects a string in the value field, so it never input other types, and I can't send `queryParamToType<T>(baseType, [value]);`.  I'm not sure if the function would process a complex type anyway (final case is to return undefined)

Reply:
i mean, the issue is if i have a setting of `t.array(t.number)`, this won't cast the values in the query parameters to numbers. Can we update `queryParamToType` to unwrap the array in that case?